src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs

```diff
@@ -72,7 +81,8 @@ public int OnAfterSnippetsUpdate()
 
             if (_expansionManager != null)
             {
-                PopulateSnippetCaches();
+                var token = _waiter.BeginAsyncOperation(GetType().Name + ".Start");
+                PopulateSnippetCachesAsync().CompletesAsyncOperation(token).Forget();
```

Comment:
Ditto: consider adding these to a JoinableTaskCollection or the like to avoid random running later.